Earliear zfs patch called nautilus_window_slot_stop_loading. 
It is not safe and can lead to core dumps in 
nautilus_notebook_sync_loading. So we just redo what 
zfs-related part of nautilus_window_slot_stop_loading should 
have done.
Perhaps it's not an ideal fix, but this heals nautilus crash
on closing tab.

--- nautilus-2.32.2.1/src/nautilus-window-manage-views.c.1	2015-05-18 13:18:37.086018624 +0300
+++ nautilus-2.32.2.1/src/nautilus-window-manage-views.c	2015-05-18 13:20:08.230749028 +0300
@@ -2092,11 +2092,18 @@
 nautilus_window_manage_views_close_slot (NautilusWindowPane *pane,
 					 NautilusWindowSlot *slot)
 {
+	NautilusDirectory *directory;
+
 	if (slot->content_view != NULL) {
 		nautilus_window_slot_disconnect_content_view (slot, slot->content_view);
 	}
 
-	nautilus_window_slot_stop_loading (slot);
+	if (slot->find_zfs_snapshots_cancellable)
+		g_cancellable_cancel (slot->find_zfs_snapshots_cancellable);
+
+	directory = nautilus_directory_get (slot->location);
+	nautilus_directory_cancel_restore_info (directory);
+	nautilus_directory_unref (directory);
 
 	free_location_change (slot);
 	cancel_viewed_file_changed_callback (slot);
